src/sentry/search/events/builder/metrics.py

```diff
@@ -112,30 +116,32 @@ def _resolve_on_demand_spec(
             return None
 
     def _get_on_demand_metrics_query(self) -> Optional[MetricsQuery]:
-        if not self.is_performance or not self.is_alerts_query:
-            return None
-
         spec = self._on_demand_spec
 
+        # TimeseriesQueryBuilder specific parameters
+        if self.is_timeseries_query:
```

Comment:
Wouldn't it be easier to just check `if isinstance(self, TimeSeriesQueryBuilder)`

Reply:
Yes! Done in [25d8adb](https://github.com/getsentry/sentry/pull/53053/commits/25d8adb1da7d5cfa6cb7a80ed25ed2cdb0eccddc)